src/main/java/hobbylist/logic/parser/FindCommandParser.java

```diff
@@ -25,9 +31,136 @@ public FindCommand parse(String args) throws ParseException {
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
 
-        String[] nameKeywords = trimmedArgs.split("\\s+");
+        String[] keywords = trimmedArgs.split("\\s+");
 
-        return new FindCommand(new NameOrDescContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+        String time = DEFAULT_DATE;
+        int value = DEFAULT_RATING;
+
+        if (isValidKeywords(keywords)) {
+            time = getDate(keywords);
+            value = getRating(keywords);
+        }
+
+        return new FindCommand(new NameOrDescContainsKeywordsPredicate(Arrays.asList(keywords)),
+                new DateMatchesGivenDatePredicate(time),
+                new RatingMatchesGivenValuePredicate(value));
+    }
+
+    private static boolean isValidKeywords(String[] keywords) throws ParseException {
+        if (hasDateKeyword(keywords)) {
+            if (Arrays.stream(keywords).filter(FindCommandParser::isDateKeyword).count() > 1) {
```

Comment:
Would it be better to search for multiple tags or ratings when multiple are provided?